packages/openapi-generator/src/options/generator-options.ts

```diff
@@ -0,0 +1,40 @@
+import Parser from '@oclif/parser';
+import GenerateOpenApiClient from '../cli';
+
+export interface GeneratorOptions {
+  input: string;
+  outputDir: string;
+  transpile?: boolean;
+  include?: string;
+  clearOutputDir?: boolean;
+  strictNaming?: boolean;
+  tsConfig?: string;
+  packageJson?: boolean;
+  perServiceConfig?: string;
+  packageVersion?: string;
+  readme?: boolean;
+  metadata?: boolean;
+}
+
+export type ParsedGeneratorOptions = typeof GenerateOpenApiClient extends Parser.Input<
+  infer F
+>
+  ? F
+  : never;
+
+export function parseGeneratorOptions(
```

Comment:
a doc would help. Is this needed? Should oclif not set the default?

Reply:
See my other comments. I added documentation as well 😊